packages/database/src/modelClasses/Entity.js

```diff
@@ -443,11 +443,11 @@ export class EntityModel extends MaterializedViewLogDatabaseModel {
           sort: ['generational_distance ASC'],
         },
       );
-      return Promise.all(relations.map(async r => r.getData()));
+      const relationData = await Promise.all(relations.map(async r => r.getData()));
+      const uniqueEntities = Object.values(keyBy(relationData, 'id'));
```

Comment:
I realise this is not a new thing, but it seems a shame not to have guaranteed order preservation here when the db query has a sort in it

Reply:
Ahh that's a great point, hadn't thought of that... I think it should still preserve order though? I think lodash keyby will insert the keys as per the order they appear in the original array. I was testing with the entity-server then and it seems they still come out in generational distance order 😄

Reply:
Javascript objects don't guarantee any order, so it's possible we could see the order change unexpectedly, but in practice yeah we probably will end up with preserved order, just because that's the easiest thing for the lazy-bones computer to do 😛

Reply:
I think as of ES2015 the non integer like keys are kept in insertion order, and thus should consistently be correct? @kael89 loves to remind me of this as I have made similar statements recently too 😝

Reply:
Oh I totally remember @kael89 mentioning this to me too now! Thanks for tagging him in here to throw me under the bus haha (sorry for forgetting Kostas!!)